internal/handlers/handlers.go

```diff
@@ -0,0 +1,549 @@
+package handlers
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	nd "github.com/fedgolang/go_final_project/internal/lib/nextdate"
+	"github.com/fedgolang/go_final_project/internal/storage"
+	"github.com/golang-jwt/jwt"
+)
+
+var (
+	limitForTasks = 50                                                                         // Максимальное кол-во возвращаемых тасков в GetTasks
+	JWTSecret     = []byte("69612fb755d66b4a275896981874c46210f4afbac7673bcb0ce40d3c6a0160d5") // Секрет для токена
+)
+
+type SignInRequest struct {
+	Password string `json:"password"`
+}
+
+type SignInResponse struct {
+	Token string `json:"token,omitempty"`
+	Err   string `json:"error,omitempty"`
+}
+
+// Структура для ответа после POST
+type Response struct {
+	ID  int    `json:"id,omitempty"`
+	Err string `json:"error,omitempty"`
+}
+
+// Структура для ответа GET запроса
+// Так как задач может быть много, то у нас массив респонсов
+type TasksResponse struct {
+	Tasks []storage.TaskNoEmpty `json:"tasks"`
+}
+
+// Для реализации всех хендлеров будем пользоваться middleware
+
+// Хендлер отвечает за добавление таски в БД
+func PostTask(s *storage.Scheduler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		task := storage.Task{}
+		resp := Response{}
+		var buf bytes.Buffer
+
+		// Читаем данные из тела и запишем в буфер
+		_, err := buf.ReadFrom(r.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest) // Вернём 400 если не смогли прочитать боди
+			return
+		}
+
+		// Расшифруем JSON
+		err = json.Unmarshal(buf.Bytes(), &task)
+		if err != nil {
+			resp.Err = "Ошибка десериализации JSON"
+		}
+
+		// Проверим, что заголовок не пустой
+		if task.Title == "" {
+			resp.Err = "Не указан заголовок задачи" // 400 пришел пустой title
+		}
+
+		// Проверим, что дата не пустая
+		if task.Date == "" {
+			task.Date = time.Now().Format("20060102")
+		}
+
+		date, err := time.Parse("20060102", task.Date)
+		if err != nil {
+			resp.Err = fmt.Sprint(err)
+		}
+
+		// Проверим, что дата не прошлое, если повторения нет
+		if task.Repeat == "" {
+			if date.Before(time.Now()) {
+				task.Date = time.Now().Format("20060102")
+			}
+		} else { // Если не пустое повторение, вычислим следующую дату из NextDate()
+			// Доп проверка, если таска добавляется сегодня с датой > сегодня
+			// То не учитываем NextDate и регаем таску с датой = дате создания
+			if date.Truncate(24 * time.Hour).Before(time.Now().Truncate(24 * time.Hour)) {
+				task.Date, err = nd.NextDate(time.Now(), task.Date, task.Repeat)
+				if err != nil {
+					resp.Err = fmt.Sprint(err)
+				}
+			}
+		}
+
+		// Запись проводит только в случае отсутствия ошибки
+		if resp.Err == "" {
+			id, err := s.PostTask(task)
+			if err != nil {
+				if err == fmt.Errorf("неверный формат repeat") {
+					resp.Err = "Правило повторения указано в неправильном формате"
+				} else {
+					resp.Err = fmt.Sprint(err)
```

Comment:
Если я правильно понял идею
```suggestion
					resp.Err = err.Error()
```